### DISCUSSION TITLE
The request requires valid record authorization token to be set

### DISCUSSION THREAD

**Discussion by renatocaliari** [pocketbase/pocketbase#2811] [Q&A] [answered]
After calling `confirmVerification` and passing a valid token successfully, I encountered an error when calling `confirmVerification(token)`.
![image](https://github.com/pocketbase/pocketbase/assets/1719651/7ffef176-2bd6-407e-9e51-6f06cc79cc8a)

What could I be doing wrong? Interestingly, apparently this same code was working before.

**Comment by ganigeorgiev** [comment]
I'm not sure I follow. The screenshot seems to show `/auth-refresh` call, not `/confirm-verification`.
Note that the verification token is different from the auth token. The verification token can be extracted from the url that was clicked in the email.

**Reply by renatocaliari** [reply]
Sorry, my mistake in writing. The error is when calling authRefresh.

My code looks like this:
```
	await pb.collection('users').confirmVerification(token);
	await pb.collection('users').authRefresh();
```

The first line is executed successfully. The second one throws an error.

**Reply by ganigeorgiev** [reply, marked as answer]
Are you sure that you are authenticated?

Note that `confirmVerification()` returns blank 204 and doesn't perform authentication (it signalizes the backend to update the `verified` state of the token related auth record).

`authRefresh()` will work only if you are already authenticated.

**Reply by renatocaliari** [reply]
After clicking on the confirmation email link, does the person need to log in with email and password? 

I was assuming that the person would already be automatically logged in with that code.
Currently, after clicking on the verification/confirmation email, I try to execute that code with the token.

**Comment by renatocaliari** [comment]
I think I understand now (hopefully). The `authrefresh` only makes sense if there is already a logged-in user (authenticated).
So I modified my code. If there is no logged-in user, I redirect to the login screen.
